core/startup/config_file_watcher.go

```diff
@@ -73,11 +73,13 @@ func WatchConfigDirectory(configDir string, appConfig *config.ApplicationConfig)
 				if !ok {
 					return
 				}
-				if event.Has(fsnotify.Write) {
+				if event.Has(fsnotify.Write) || event.Has(fsnotify.Create) || event.Has(fsnotify.Rename) {
 					for targetName, watchFn := range CONFIG_FILE_UPDATES {
-						if event.Name == targetName {
+						if filepath.Base(event.Name) == targetName {
 							err := watchFn(configDir, appConfig)
-							log.Warn().Msgf("WatchConfigDirectory goroutine for %s: failed to update options: %+v", targetName, err)
+							if err != nil {
+								log.Warn().Msgf("WatchConfigDirectory goroutine for %s: failed to update options: %+v", targetName, err)

```

Comment:
we should leverage the logging facility for the metadata, something like:
`log.Error().Err(err).Str("filename", targetName).Msg("WatchConfigDirectory goroutine failed to update options")`